src/main/java/org/ambraproject/rhino/service/impl/ArticleCrudServiceImpl.java

```diff
@@ -594,4 +617,68 @@ public void updatePreprintDoi(ArticleIngestionIdentifier articleId, String prepr
     articleIngestion.setPreprintDoi(preprintOfDoi);
     hibernateTemplate.save(articleIngestion);
   }
+
+  @Override
+  public Collection<String> getArticleDois(int pageNumber, int pageSize, SortOrder sortOrder) {
+    final Collection<String> articleDois = getArticleDoisForDateRange(
+        pageNumber, pageSize, sortOrder, Optional.empty() /* fromDate */,
+        Optional.empty() /* toDate */);
+    return articleDois;
+  }
+
+  @Override
+  public Collection<String> getArticleDoisForDateRange(
+      int pageNumber, int pageSize, SortOrder sortOrder, Optional<LocalDateTime> fromDate,
+      Optional<LocalDateTime> toDate) {
+    final long totalArticles = hibernateTemplate.execute(session -> {
+      final Query query = session.createQuery("select count(*) from Article");
+      final Long count = (Long) query.uniqueResult();
+      return count;
+    });
+
+    if (totalArticles > 0L) {
+      pageNumber = max(pageNumber, 1);
+      final int maxResults = min(pageSize, MAX_PAGE_SIZE);
+      final int firstResult = (pageNumber - 1) * maxResults;
+
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("pageNumber: {}, pageSize: {}", pageNumber, pageSize);
+        LOG.debug("firstResult: {}, maxResults: {}", firstResult, maxResults);
+        LOG.debug("sortOrder: {}", sortOrder);
+      }
+
+      if (firstResult < totalArticles) {
+        final DetachedCriteria criteria = DetachedCriteria.forClass(Article.class);
+        final ProjectionList projections = Projections.projectionList();
+        projections.add(Projections.property("doi" /* propertyName */));
+        criteria.setProjection(projections);
+
+        // Set restrictions for filtering on date range, if any.
+        if (fromDate.isPresent() && toDate.isPresent()) {
+          criteria.add(Restrictions.ge(
+              "created" /* propertyName */, java.sql.Timestamp.valueOf(fromDate.get())));
+          criteria.add(Restrictions.le(
+              "created" /* propertyName */, java.sql.Timestamp.valueOf(toDate.get())));
+        } else if (fromDate.isPresent()) {
+          criteria.add(Restrictions.ge(
+              "created" /* propertyName */, java.sql.Timestamp.valueOf(fromDate.get())));
+        } else if (toDate.isPresent()) {
+          criteria.add(Restrictions.le(
+              "created" /* propertyName */, java.sql.Timestamp.valueOf(toDate.get())));
+        }
```

Comment:
Can this if simplified as
if (fromDate...) {
  ...
}
if (toDate...) {
  ...
}

Reply:
Nice catch :+1: Done.